realm/realm-library/src/main/java/io/realm/RealmSchema.java

```diff
@@ -49,7 +49,7 @@
     // Caches Class objects (both model classes and proxy classes) to their Schema object
     private final Map<Class<? extends RealmModel>, RealmObjectSchema> classToSchema = new HashMap<>();
     // Caches Class Strings to their Schema object
-    private final Map<String, RealmObjectSchema> dynamicClassToSchema = new HashMap<>();
+    final Map<String, RealmObjectSchema> dynamicClassToSchema = new HashMap<>();
```

Comment:
I suggest, then, that you add `protected` methods to the superclass, that control the kinds of modifications that the subclass can perform.  A good rule is "`private` or `immutable`".  That is, even, in our lint config.